server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceRegistry.java

```diff
@@ -152,4 +168,8 @@ public ValuesSourceType getValuesSourceType(MappedFieldType fieldType, String ag
             }
         }
     }
+
+    public UsageService getUsageService() {
```

Comment:
I made a note below about documenting the concurrency behavior, but we should also call it out here.  In particular, nothing else we return out of the registry has a mutable internal state, but this does.